packages/itwinui-css/src/input/input-flex-container.scss

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@mixin iui-input-flex-container {
+  --_iui-input-background-color: var(--iui-color-background);
+  --_iui-input-border-color: var(--iui-color-border-foreground);
+  --_iui-input-text-color: var(--iui-color-text);
+
+  @include iui-reset;
+  @include iui-focus($offset: -2px, $thickness: 2px);
+  width: 100%;
+  font-family: inherit;
+  font-size: var(--iui-font-size-1);
+  font-weight: var(--iui-font-weight-normal);
+  line-height: var(--iui-size-l);
+  border-radius: var(--iui-border-radius-1);
+  appearance: none;
+  height: var(--_iui-input-min-height);
+  padding-block: var(--_iui-input-padding-block);
+  padding-inline: var(--iui-size-s);
+  color: var(--_iui-input-text-color);
+  background-color: var(--_iui-input-background-color);
+  border: 1px solid var(--_iui-input-border-color);
+  transition: border-color var(--iui-duration-1) ease-out;
+  @include iui-input-size;
+
+  right: unset;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: 0;
+  margin: 0;
+
+  > input {
+    padding-left: var(--iui-size-s);
+    width: 100%; // Take all available space
+    color: var(--iui-color-text);
+    font-family: inherit;
+    font-size: var(--iui-font-size-1);
+    font-weight: var(--iui-font-weight-normal);
+    line-height: var(--iui-size-l);
+
+    &::placeholder {
+      --_iui-input-text-color: var(--iui-color-text-disabled);
+      user-select: none;
+      color: var(--_iui-input-text-color);
+    }
+  }
+
+  &:autofill,
+  > input:autofill {
+    --_iui-input-background-color: var(--iui-color-background-accent-muted);
+    --_iui-input-border-color: var(--iui-color-border-accent);
+    --_iui-input-text-color: var(--iui-color-text-accent);
+  }
+
+  &[disabled],
+  > input[disabled] {
+    --_iui-input-background-color: var(--iui-color-background-disabled);
+    --_iui-input-border-color: var(--iui-color-border-disabled);
+    --_iui-input-with-icon-hover-border-color: var(--iui-color-border-disabled);
+    cursor: not-allowed;
+  }
+
+  > .iui-svg-icon {
```

Comment:
should not directly target other components here. I also wonder if we need to offer a padded version of the icon.

but for now, you can set the svg inside CSS as mentioned in my [other comment](https://github.com/iTwin/iTwinUI/pull/863#discussion_r1097980519)

Reply:
Targeting only input - I think I could create new class for that to apply styles.
Also targeting collapsed/expanded states in searchbox.scss but you'll have to check if this approach is good enough for you - might need to rework that.